### ISSUE TITLE
[BUG] Strange L1 metrics on Microsoft hardware

### ISSUE THREAD

**Issue by mcandre** [google/benchmark#1133]
**Describe the bug**

When I lookup L1 cache metrics on Microsoft Surface Pro 7, I get conflicting information. Google benchmark says one thing and Task Manager says another.

**System**
Which OS, compiler, and compiler version are you using:
  - OS: Windows 10 and Ubuntu 20.04 on WSL 2.
  - Compiler and version: clang 10 and MSVC 19.28.

**To reproduce**

![task-manager](https://user-images.githubusercontent.com/5316/116767310-06237200-a9f5-11eb-8553-5710194894da.png)

![google-benchmark-native](https://user-images.githubusercontent.com/5316/116767315-0b80bc80-a9f5-11eb-9460-083873cfb391.png)

![google-benchmark-wsl](https://user-images.githubusercontent.com/5316/116767316-0e7bad00-a9f5-11eb-9291-ed597009aaf6.png)

Source: https://github.com/mcandre/swiper

**Expected behavior**

All the L1 cache numbers should match up. Normally L1 cache is split right in half between instructions and data. Or at least adds up to the unified L1 size.

**Comment by pleroy** [comment]
The numbers seem pretty consistent to me:
```
L1: 4 * (48 KiB + 32 KiB) = 320 KiB
L2: 4 * 512 KiB = 2 MiB
L3: 1 * 8192 KiB = 8 MiB
```
The L2 and L3 numbers reported by the library also match those on the [Wikipedia page for i7 Sunny Cove](https://en.wikipedia.org/wiki/List_of_Intel_Core_i7_processors#Sunny_Cove_microarchitecture_(10th_generation)).  As regards the unsupported theory that the L1 cache is split in half, [this page](https://www.cpu-world.com/CPUs/Core_i7/Intel-Core%20i7%20i7-1065G7.html) seems to indicate that it's not the case.

The bottom line is that the library looks correct, and maybe the feature request should be on Microsoft to report more precise data.
